src/index.js

```diff
@@ -3294,60 +3304,54 @@ async function retrieveBackgrounds() {
 }
 
 /**
- * Retrieve metadata for files in the play queue
+ * Retrieve metadata for the first MAX_METADATA_REQUESTS files in the play queue,
+ * which have no metadata assigned yet
  */
 async function retrieveMetadata() {
-	// leave when we already have enough concurrent requests pending
-	if ( waitingMetadata >= MAX_METADATA_REQUESTS )
-		return;
-
-	// find the first play queue item for which we haven't retrieved the metadata yet
-	const queueItem = Array.from( playlist.children ).find( el => el.dataset.retrieve );
 
-	if ( queueItem ) {
+	// find the first MAX_METADATA_REQUESTS items for which we haven't retrieved the metadata yet
+	const retrievalQueue = Array.from(playlist.children).filter(el => el.dataset.retrieve).slice(0, MAX_METADATA_REQUESTS);
 
+	// Execute in parallel
+	return Promise.all(retrievalQueue.map(async queueItem => {
 		let uri    = queueItem.dataset.file,
 			revoke = false;
 
-		waitingMetadata++;
 		delete queueItem.dataset.retrieve;
 
-		queryMetadata: {
-			if ( queueItem.handle ) {
-				try {
-					if ( await queueItem.handle.requestPermission() != 'granted' )
-						break queryMetadata;
 
-					uri = URL.createObjectURL( await queueItem.handle.getFile() );
-					revoke = true;
-				}
-				catch( e ) {
-					break queryMetadata;
-				}
+		if ( queueItem.handle ) {
+			try {
+				if ( await queueItem.handle.requestPermission() !== 'granted' )
+					return;
+
+				uri = URL.createObjectURL( await queueItem.handle.getFile() );
+				revoke = true;
+			}
+			catch( e ) {
+				consoleLog(`Error converting queued file="${queueItem.handle.file}" to URI`, e);
+				return;
 			}
+		}
 
-			try {
-				const metadata = await mm.fetchFromUrl( uri, { skipPostHeaders: true } );
-				if ( metadata ) {
-					addMetadata( metadata, queueItem ); // add metadata to play queue item
+		try {
+			const metadata = await mm.fetchFromUrl( uri, { skipPostHeaders: true } );
```

Comment:
Did you inadvertently revert this? It's still using `fetchFromUrl()`. I think this is now dependent on changes in #89